app/models/user.rb

```diff
@@ -416,6 +416,31 @@ def closed?
     closed_at.present?
   end
 
+  def erase
+    erase!
+  rescue ActiveRecord::RecordInvalid, ArgumentError
+    false
+  end
+
+  def erase!
+    raise ArgumentError unless closed?
```

Comment:
Not sure `ArgumentError` is the best exception to raise here. Raising `ActiveRecord::RecordInvalid` or our own `RecordInvalid` exception would be better imo